src/super_gradients/common/crash_handler/crash_tips.py

```diff
@@ -0,0 +1,209 @@
+        factory_name, _ = RecipeFactoryFormatTip._get_factory_with_params(exc_value)
+        factory_name = fmt_txt(factory_name, bold=True, color="green")
+        msg = f"There is an indentation error in the recipe, while creating {factory_name}."
+        return msg
+
+    @classmethod
+    def _get_solution(cls, exc_type: type, exc_value: Exception, exc_traceback: TracebackType) -> str:
+        factory_name, params_dict = RecipeFactoryFormatTip._get_factory_with_params(exc_value)
+
+        params_in_yaml = "\n".join(f"  {k}: {v}" for k, v in params_dict.items())
+        user_yaml = f"- {factory_name}:\n" + params_in_yaml
+        user_yaml = fmt_txt(user_yaml, indent=4, color="red")
+
+        correct_yaml = f"- {factory_name}:\n" + indent_string(params_in_yaml, indent_size=2)
+        correct_yaml = fmt_txt(correct_yaml, indent=4, color="green")
+        msg = "If your yaml looks like this:\n" f"{user_yaml}\n" f"Then you change your recipe to this:\n" f"{correct_yaml}"
+
+        return msg
+
+    @staticmethod
+    def _get_factory_with_params(exc_value: Exception) -> Tuple[str, dict]:
+        """Utility function to extract useful features from the exception.
+        :return: Name of the factory that (we assume) was not correctly defined
+        :return: Parameters that are passed to that factory
+        """
+        description = str(exc_value)
+        params_dict = re.search(r"received: (.*?)$", description).group(1)
+        params_dict = json_str_to_dict(params_dict)
+        factory_name = next(iter(params_dict))
+        params_dict.pop(factory_name)
+        return factory_name, params_dict
+
+
+class DDPNotInitializedTip(CrashTip):
+    """Note: I think that this should be caught within the code instead"""
+
+    @classmethod
+    def is_relevant(cls, exc_type: type, exc_value: Exception, exc_traceback: TracebackType):
+        expected_str = "Default process group has not been initialized, please make sure to call init_process_group."
+        return isinstance(exc_value, RuntimeError) and expected_str in str(exc_value)
+
+    @classmethod
+    def _get_explanation(cls, exc_type: type, exc_value: Exception, exc_traceback: TracebackType) -> str:
+        msg = "Your environment was not setup correctly for DDP."
+        return msg
+
+    @classmethod
+    def _get_solution(cls, exc_type: type, exc_value: Exception, exc_traceback: TracebackType) -> str:
+        msg = (
+            "Please run at the beginning of your script:\n"
+            f">>> {fmt_txt('from super_gradients.training.utils.distributed_training_utils import setup_gpu_mode', color='green')}\n"
+            f">>> {fmt_txt('from super_gradients.common.data_types.enum import MultiGPUMode', color='green')}\n"
+            f">>> {fmt_txt('setup_gpu_mode(gpu_mode=MultiGPUMode.DISTRIBUTED_DATA_PARALLEL, num_gpus=...)', color='green')}"
+        )
+        return msg
```

Comment:
Question:
This tip aims at fixing an error which is predictable when the user params are wrong.
Do we want to add a check and raising a custom error within SG, or to leave it as is and add a crash tip at the end ?

Reply:
If you can catch it and raise a good error, then yes. but make sure you are covering all cases, not just the main one